test/dynamo/test_activation_checkpointing.py

```diff
@@ -519,6 +519,7 @@ def _factory_fn():
                 mod_no_hook, backend, x, fullgraph=True, compiled_autograd=True
             )
 
+        torch._dynamo.reset()
```

Comment:
@yf225 From what I understand, this test was written to check that the partitioner is able to save the new intermediate from the backward hook.

However, the way the test is written, currently it relies on Dynamo to see that a new hook has been applied to the module and recompile. With the change in this PR, I am bringing some unsoundness because this gives quite good speedup in general. So, I am wondering if this is a common case where you compile the model and then apply the hook? Or this is just a co-incidence and was not really the intention of this test.

Reply:
I believe it's just a co-incidence of how the test is written and was not the intention of this test (it's more meant to have an entirely new module to register a hook on).

I believe as long as the check `self.assertEqual(additional_saved_tensors, expected_additional_saved_tensors)` can still pass (which relies on the Inductor post_grad pass `collect_fwd_graph_outputs` to collect the saved tensors) we are good here